ci_watson/artifactory_helpers.py

```diff
@@ -125,6 +147,309 @@ def get_bigdata(*args):
+
+    Example:
+    This example from an actual test definition demonstrates
+    how multiple input defintions can be used at the same time.::
+
+        outputs = [( # Compare psfstack product
+                    'jw99999-a3001_t1_nircam_f140m-maskbar_psfstack.fits',
+                    'jw99999-a3001_t1_nircam_f140m-maskbar_psfstack_ref.fits'
+                   ),
+                   (
+                    'jw9999947001_02102_00002_nrcb3_a3001_crfints.fits',
+                    'jw9999947001_02102_00002_nrcb3_a3001_crfints_ref.fits'
+                   ),
+                   {'files':( # Compare i2d product
+                            'jw99999-a3001_t1_nircam_f140m-maskbar_i2d.fits',
+                            'jw99999-a3001_t1_nircam_f140m-maskbar_i2d_ref.fits'
+                     ),
+                     'pars': {'ignore_hdus':self.ignore_hdus+['HDRTAB']}
+                   },
+                   {'files':( # Compare the HDRTAB in the i2d product
+                    'jw99999-a3001_t1_nircam_f140m-maskbar_i2d.fits[hdrtab]',
+                    'jw99999-a3001_t1_nircam_f140m-maskbar_i2d_ref.fits[hdrtab]'
+                   ),
+                    'pars': {'ignore_keywords':
+                             self.ignore_keywords+['NAXIS1', 'TFORM*'],
+                             'ignore_fields':self.ignore_keywords}
+                   }
+                  ]
+    .. NOTE::
+    Note that each entry in the list gets interpreted and processed
+    separately.
+    """
+    all_okay = True
+    creature_report = ''
+    # Create instructions for uploading results to artifactory for use
+    # as new comparison/truth files
+    testpath, testname = os.path.split(os.path.abspath(os.curdir))
+    # organize results by day test was run...could replace with git-hash
+    whoami = getpass.getuser() or 'nobody'
+    dt = datetime.datetime.now().strftime("%d%b%YT")
+    ttime = datetime.datetime.now().strftime("%H_%M_%S")
+    user_tag = 'NOT_CI_{}_{}'.format(whoami, ttime)
+    build_tag = os.environ.get('BUILD_TAG',  user_tag)
+    build_suffix = os.environ.get('BUILD_MATRIX_SUFFIX', 'standalone')
+    testdir = "{}_{}_{}".format(testname, build_tag, build_suffix)
+
+    # Parse any user-specified kwargs
+    ignore_keywords = kwargs.get('ignore_keywords', [])
+    ignore_hdus = kwargs.get('ignore_hdus', [])
+    ignore_fields = kwargs.get('ignore_fields', [])
+    rtol = kwargs.get('rtol', None)
+    atol = kwargs.get('atol', None)
+    input_path = kwargs.get('input_path', [])
+    results_root = kwargs.get('results_root', None)
+    docopy = kwargs.get('docopy', True)
+
+    updated_outputs = []
+    extn_list = None
+    for entry in outputs:
+        num_entries = len(entry)
+        if isinstance(entry, dict):
+            actual = entry['files'][0]
+            desired = entry['files'][1]
+            diff_pars = entry['pars']
+            ignore_keywords = diff_pars.get('ignore_keywords', ignore_keywords)
```

Comment:
@stsci-hack , I *think* this is a bug because this logic overwrites what is provided by `kwargs` permanently. I'll think of a fix but please let me know if I misunderstood.

Reply:
From all my testing and investigation, kwargs always remains unchanged.  Only variables change their value based on what is either in `kwargs` or `diff_pars`.  Now, I am probably not exactly clear on what you see as a problem, so more information would be helpful.

Reply:
This logic:
```
    for entry in outputs:
        if isinstance(entry, dict):
            ...
            ignore_keywords = diff_pars.get('ignore_keywords', ignore_keywords)
            ignore_hdus = diff_pars.get('ignore_hdus', ignore_hdus)
            ignore_fields = diff_pars.get('ignore_fields', ignore_fields)
            rtol = diff_pars.get('rtol', rtol)
```
When `entry` is a dictionary, whatever obtained from `kwargs` is overwritten by `diff_pars`. This change is not reverted if the next entry is a list and not a dictionary. Since your documentation suggests that `outputs = [list, list, dict, list, dict, list]` is acceptable, this looks like a bug to me.

Reply:
Thank you for the clarification.  You are correct that this would result in values being used for subsequent entry's that may not be correct.  I will see what can be done to resolve this logic problem.

Reply:
Warren, no worries. I think I have a fix.

Reply:
Actually, take a look at 98ae500 to see whether this will do.